### PR TITLE
Fix additon of extraneous fields

### PR DESCRIPTION
**Pull request by lishaduck**
Haven't tracked down the source of the bug yet.

### CODE REVIEW

File: src/__tests__/type-syncer.test.ts [hunk 1]
```diff
@@ -161,20 +168,11 @@ function buildSyncer() {
...
-          workspaces = package2File.workspaces
-          break
-        }
         default:
           throw new Error('What?!')
       }
 
-      workspaces ??= []
-      const globPromises = workspaces.map((w) =>
+      const globPromises = workspaces!.map((w) =>
```
Comment (lishaduck) [inline src/__tests__/type-syncer.test.ts:175]: Yes, non-null assertations are bad, but I figured they fail fast, unlike `??=`.
Same thing with the mocks.

**Comment by lishaduck** [comment]
I gave up writing regression tests, but otherwise I'm happy with this.

**Comment by lishaduck** [comment]
Also, if developing on node>=18, you can apply a patch and build with `tsdown`, which is much faster:

<details><summary>Patch</summary>
<p>

```diff ts
diff --git a/bin/typesync b/bin/typesync
index 95dbe55..e28f57c 100755
--- a/bin/typesync
+++ b/bin/typesync
@@ -1,4 +1,2 @@
 #!/usr/bin/env node
-const { startCli } = require('../lib/cli')
-
-startCli()
+import('../dist/cli.mjs').then(({ startCli }) => startCli())
```
</p>
</details> 

Build command: `pnpm dlx tsdown --watch src/cli.ts` (`npx tsdown --watch src/cli.ts` for npm)

**Comment by jeffijoe** [comment]
I’m out on vacation, I’ll check on these when I get back. 👍

File: src/type-syncer.ts [hunk 2]
```diff
@@ -65,7 +65,14 @@ export function createTypeSyncer(
 
     const syncedFiles: Array<ISyncedFile> = await Promise.all([
       syncFile(filePath, file, syncOpts, dryRun),
-      ...subManifests.map((p) => syncFile(p, null, syncOpts, dryRun)),
+      ...subManifests.map(async (p) =>
+        syncFile(
+          p,
+          await packageJSONService.readPackageFile(p),
```
Comment (jeffijoe) [inline src/type-syncer.ts:71]: Just curious, what did this change vs just reading the file from the file path in `syncFile`?

Comment (lishaduck) [reply src/type-syncer.ts:71]: I honestly don't remember 🤷‍♂️

**Pull request merged**